cdap-master/src/main/java/co/cask/cdap/data/tools/DatasetUpgrader.java

```diff
@@ -93,45 +96,40 @@ private void upgradeSystemDatasets(DatasetFramework framework) throws Exception
   }
 
   private void upgradeUserTables() throws Exception {
-    DefaultDatasetNamespace namespace = new DefaultDatasetNamespace(cConf);
     HBaseAdmin hAdmin = new HBaseAdmin(hConf);
 
-
-    for (HTableDescriptor desc : hAdmin.listTables()) {
+    for (HTableDescriptor desc : hAdmin.listTables(USER_TABLE_PREFIX)) {
       String tableName = desc.getNameAsString();
-      TableId tableId = TableId.from(tableName);
-      Id.DatasetInstance datasetInstanceId = Id.DatasetInstance.from(tableId.getNamespace(), tableId.getTableName());
-      // todo: it works now, but we will want to change it if namespacing of datasets in HBase is more than +prefix
-      if (namespace.fromNamespaced(datasetInstanceId) != null) {
-        LOG.info("Upgrading hbase table: {}, desc: {}", tableName, desc);
-
-        final boolean supportsIncrement = HBaseTableAdmin.supportsReadlessIncrements(desc);
-        final boolean transactional = HBaseTableAdmin.isTransactional(desc);
-        DatasetAdmin admin = new AbstractHBaseDataSetAdmin(tableId, hConf, hBaseTableUtil) {
-          @Override
-          protected CoprocessorJar createCoprocessorJar() throws IOException {
-            return HBaseTableAdmin.createCoprocessorJarInternal(cConf,
-                                                                locationFactory,
-                                                                hBaseTableUtil,
-                                                                transactional,
-                                                                supportsIncrement);
-          }
-
-          @Override
-          protected boolean upgradeTable(HTableDescriptor tableDescriptor) {
-            // we don't do any other changes apart from coprocessors upgrade
-            return false;
-          }
-
-          @Override
-          public void create() throws IOException {
-            // no-op
-            throw new UnsupportedOperationException("This DatasetAdmin is only used for upgrade() operation");
-          }
-        };
-        admin.upgrade();
-        LOG.info("Upgraded hbase table: {}", tableName);
-      }
+      HTableNameConverter hTableNameConverter = new HTableNameConverterFactory().get();
```

Comment:
Actually mis-understood your question. This should be ok, since ``HTableNameConverter`` has logic for maintaining backward-compatibility, etc specific to an HBase version.